File: elixir/word-count/word_count.exs
defmodule Words do
  @doc """
  Count the number of words in the sentence.

  Words are compared case-insensitively.
  """
  @spec count(String.t) :: map
  def count(sentence) do
    sentence
    |> String.downcase
    |> String.split( ~r{\s|_|,|\W\W} , trim: true)
    |> Enum.reduce(%{}, fn(i, acc) -> Map.update(acc, i, Map.get(acc, i, 0) + 1, fn (x) -> x + 1 end ) end )
  end
end

File: python/word-count/word_count.py
import re

def word_count(sentence):
    
    words = {}

    for word in re.split('[^a-zA-Z0-9]', sentence.lower()):
        if word != "":
            if word in words:
                words[word] = words[word] + 1
            else:
                words[word] = 1

    return words
